### PR TITLE
feat: Implementation of Hospital service

### PR DESCRIPTION
**Pull request by Driemtax**
My Implementation for the hospital service. There are 2 major tasks the service has to complete:

Firstly it needs to update the humans and hospitals according to the patient that are already in the hospital. So it needs to check if a patient has died in hospital or if a patient is healthy again and can go back home. 

Secondly it needs to check if new patients are waiting in front of the hospital. Therefore it checks if someone is at the hospital, but not hospitalized yet, has the goal hospital and is infected. In that case, if the hospital has capacity left, the human will become a patient in the hospital. 

In order to complete those 2 tasks i needed to make the GoalState enum publicly available. I also needed to alter the attributes of the human slightly. 

In my opinion all cases are checked. If you notice a case that has not yet been considered, please let me know. Same goes for any questions regarding my code.

I check for following cases:
1. A human is already in hospital but died this tick -> used hospital capacity increases, human.isHopitalized=false
2. A human is already in hospital but is healthy again -> human goes home, human not hospitalized anymore, used hospital capacity increases
3. A human is at the hospital but not patient yet, he is infected and there is capacity left -> human gets patient in hospital

### CODE REVIEW

**Comment by InaasHammoush** [comment]
I have reviewed zour code and I think that it is mostly verz good but it had brought my attention to a possible improvment in the Architecture that I would like to discuss with you and Lukas tomorrow.

**Review by InaasHammoush — COMMENTED** [review]
I have reviewed your code and I think that it is mostly very good but it had brought my attention to a possible improvment in the Architecture that I would like to discuss with you and Lukas tomorrow.

File: src/com/biocollapse/service/SimulationService.java [hunk 1]
```diff
@@ -50,4 +51,94 @@ public static LiveStatistics calculateLiveStatistics(Human[] humans, Hospital[]
...
+
+    }
+
+    /**
+     * This method checks if a human is at the same position as the hospital
+     * @param humanPos
+     * @param hospitalPos
+     * @return boolean
+     */
+    private boolean isHumanAtHospitalPos(MapPosition humanPos, MapPosition hospitalPos){
```
Comment (N3moAhead) [inline src/com/biocollapse/service/SimulationService.java]: MapPositions already have the function equals. You could call it just like this:
```java
humanPos.equals(hospitalPos)
```

So you don't have to implement this function again 👍🏻

**Review by N3moAhead — CHANGES_REQUESTED** [review]
The code looks very cleanly written and seems to do exactly what it should. I have suggested a very small improvement, but nothing major.

In general, I notice that we often have to iterate through the Humans and Hospitals. I think it would be good if we had some data structures in the initialisation of the objects and the controller, e.g. a hashmap like: HashMap<MapPosition, Hospital> or something like that to speed up certain processes during runtime. 

But we can discuss this next week 👍🏻

File: src/com/biocollapse/service/SimulationService.java (hunk 1 above)
Comment (Driemtax) [reply src/com/biocollapse/service/SimulationService.java]: I have applied these fixes

**Review by N3moAhead — APPROVED** [review]
I think it looks great now! Thanks for your work! :)

**Pull request merged**